src/main/java/de/tum/in/www1/artemis/service/scheduled/AthenaScheduleService.java

```diff
@@ -0,0 +1,111 @@
+package de.tum.in.www1.artemis.service.scheduled;
+
+import java.time.ZonedDateTime;
+import java.util.*;
+import java.util.concurrent.ScheduledFuture;
+
+import javax.annotation.PostConstruct;
+import javax.validation.constraints.NotNull;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.context.annotation.Profile;
+import org.springframework.core.env.Environment;
+import org.springframework.stereotype.Service;
+
+import de.tum.in.www1.artemis.domain.TextExercise;
+import de.tum.in.www1.artemis.domain.enumeration.ExerciseLifecycle;
+import de.tum.in.www1.artemis.repository.TextExerciseRepository;
+import de.tum.in.www1.artemis.security.SecurityUtils;
+import de.tum.in.www1.artemis.service.ExerciseLifecycleService;
+import de.tum.in.www1.artemis.service.connectors.athena.AthenaSubmissionSendingService;
+import tech.jhipster.config.JHipsterConstants;
+
+@Service
+@Profile("athena & scheduling")
+public class AthenaScheduleService {
+
+    private final Logger log = LoggerFactory.getLogger(AthenaScheduleService.class);
+
+    private final ExerciseLifecycleService exerciseLifecycleService;
+
+    private final TextExerciseRepository textExerciseRepository;
+
+    private final Environment env;
+
+    private final Map<Long, ScheduledFuture<?>> scheduledAthenaTasks = new HashMap<>();
+
+    private final AthenaSubmissionSendingService athenaSubmissionSendingService;
+
+    public AthenaScheduleService(ExerciseLifecycleService exerciseLifecycleService, TextExerciseRepository textExerciseRepository, Environment env,
+            AthenaSubmissionSendingService athenaSubmissionSendingService) {
+        this.exerciseLifecycleService = exerciseLifecycleService;
+        this.textExerciseRepository = textExerciseRepository;
+        this.env = env;
+        this.athenaSubmissionSendingService = athenaSubmissionSendingService;
+    }
+
+    @PostConstruct
+    private void scheduleRunningExercisesOnStartup() {
+        final Collection<String> activeProfiles = Arrays.asList(env.getActiveProfiles());
+        if (activeProfiles.contains(JHipsterConstants.SPRING_PROFILE_DEVELOPMENT)) {
+            // only execute this on production server, i.e. when the prod profile is active
+            // NOTE: if you want to test this locally, please comment it out, but do not commit the changes
+            return;
+        }
+        final List<TextExercise> runningTextExercises = textExerciseRepository.findAllAutomaticAssessmentTextExercisesWithFutureDueDate();
```

Comment:
Since this is the only place that uses this method, we should probably filter for `isFeedbackSuggestionsEnabled` already in the database to avoid loading to many unrelated exercises.

Please have a look at the production instance and check how many exercises will typically be returned by such a call and if it makes sense to introduce paging here.

Reply:
This already happens: The DB call fetches all exercises with assessment type `SEMI_AUTOMATIC`. Currently, that's exactly the condition that `isFeedbackSuggestionsEnabled` checks. This will change in the future (we will add an `exercise.feedback_suggestions` column in the DB), but it's just the assessment type for this PR.

However, I now noticed that the `exercise.feedbackSuggestionsEnabled` field is not needed on the client, so I've `@JsonIgnore`-d it now.

Reply:
The highest number of simultaneous `SEMI_AUTOMATIC` text exercises at once with `release_date < current_date < due_date` was in May of 2020: 4 exercises. I don't think we need paging 🎉 For reference, I used this script to find out: https://gist.github.com/pal03377/7f5eafa6c4e2900955e61713abf038fa